### PR TITLE
CI: Release start

### PR DESCRIPTION
**Pull request by chevdor**
This PR allows starting a release process as soon as the release branch is created.
It is  very close to what we do in the Polkadot repo.

The release branches are defined by the following pattern (Github does not support regexp):
```
release-**v[0-9]+.[0-9]+.[0-9]+
```

So a valid release branch will be `release-statemine-v7.0.0` for instance.

### CODE REVIEW

**Comment by chevdor** [comment]
For this workflow to work, the following issue are required:
- https://github.com/paritytech/ci_cd/issues/269
- https://github.com/paritytech/ci_cd/issues/268

File: .github/ISSUE_TEMPLATE/release.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+### Runtime Releases
+
+These checks should be performed on the codebase.
+
+- [ ] Verify [`spec_version`](#spec-version) has been incremented since the
+    last release for any native runtimes from any existing use on public
+    (non-private/test) networks.
+- [ ] Verify previously [completed migrations](#old-migrations-removed) are
+    removed for any public (non-private/test) networks. @apopiak
```
Comment (apopiak) [inline .github/ISSUE_TEMPLATE/release.md:16]:
```suggestion
- [ ] Verify previously [completed migrations](#old-migrations-removed) are
    removed for any public (non-private/test) networks.
```
The tags should not be in the template, IMO.

File: .github/ISSUE_TEMPLATE/release.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+- [ ] Verify [Polkadot JS API](#polkadot-js) are up to date with the latest
+    runtime changes.
+- [ ] Push runtime upgrade to Westmint and verify network stability.
+
+### Initial Statemint Release
+On top of the regular runtime release process, this release will also need to cover the following points:
+- [ ] genesis configuration for Statemint @apopiak
+- [ ] determine `setStorage` call for governance to set the initial state Statemint should have after the upgrade (replacing the usual genesis process, but making use of the genesis config) @apopiak
+- [ ] create adusted chain spec that will be used for syncing Statemint @NachoPal
+- [ ] test the upgrade process (including `setStorage` calls) @apopiak
```
Comment (apopiak) [inline .github/ISSUE_TEMPLATE/release.md:43]: I don't think this should be part of the general release notes, this is a one-time thing
In fact I should remove them from the v6 release

File: .github/ISSUE_TEMPLATE/release.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+- [ ] Verify that the various pieces of XCM config are sane. @apopiak
+
+The following checks can be performed after we have forked off to the release-
+candidate branch or started an additional release candidate branch (rc-2, rc-3, etc)
+
+- [ ] Verify [new migrations](#new-migrations) complete successfully, and the
+    runtime state is correctly updated for any public (non-private/test)
+    networks.
+- [ ] Verify [Polkadot JS API](#polkadot-js) are up to date with the latest
+    runtime changes.
```
Comment (apopiak) [inline .github/ISSUE_TEMPLATE/release.md:35]:
```suggestion
- [ ] Run integration tests.
  - [ ] Teleport Relay -> Statemin* and back.
  - [ ] Create asset (if applicable), mint and transfer
```

File: .github/ISSUE_TEMPLATE/release.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+### All Releases
+
+- [ ] Check that the new polkadot-collator versions have [run on the network](#burn-in)
+    without issue. @PierreBesson
+- [ ] Check that a draft release has been created at
+    https://github.com/paritytech/cumulus/releases with relevant [release
+    notes](#release-notes) @chevdor
+- [ ] Check that [build artifacts](#build-artifacts) have been added to the
+    draft-release @chevdor
```
Comment (apopiak) [inline .github/ISSUE_TEMPLATE/release.md:47]:
```suggestion
- [ ] Check that the new polkadot-collator versions have [run on the network](#burn-in)
    without issue.
- [ ] Check that a draft release has been created at
    https://github.com/paritytech/cumulus/releases with relevant [release
    notes](#release-notes).
- [ ] Check that [build artifacts](#build-artifacts) have been added to the
    draft-release.
```

File: .github/ISSUE_TEMPLATE/release.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+  - No migrations added in the last release that would need to be removed.
+- [ ] Verify pallet and [extrinsic ordering](#extrinsic-ordering) as well as `SignedExtension`s have stayed
+    the same. Bump `transaction_version` if not. @NachoPal
+- [ ] Verify new extrinsics have been correctly whitelisted/blacklisted for
+    [proxy filters](#proxy-filtering). @apopiak
+  - No new extrinsics.
+- [ ] Verify [benchmarks](#benchmarks) have been updated for any modified
+    runtime logic. @NachoPal
+  - [ ] Verify the weights are up-to-date. @NachoPal
+- [ ] Verify that the various pieces of XCM config are sane. @apopiak
```
Comment (apopiak) [inline .github/ISSUE_TEMPLATE/release.md:26]:
```suggestion
- [ ] Verify pallet and [extrinsic ordering](#extrinsic-ordering) as well as `SignedExtension`s have stayed
    the same. Bump `transaction_version` if not.
- [ ] Verify new extrinsics have been correctly whitelisted/blacklisted for
    [proxy filters](#proxy-filtering).
  - No new extrinsics.
- [ ] Verify [benchmarks](#benchmarks) have been updated for any modified
    runtime logic.
  - [ ] Verify the weights are up-to-date.
- [ ] Verify that the various pieces of XCM config are sane.
```

**Review by alvicsam — APPROVED** [review]

**Review by apopiak — APPROVED** [review]

**Pull request merged**